controller/src/beerocks/master/tasks/dynamic_channel_selection_task.h

```diff
@@ -92,8 +92,13 @@ class dynamic_channel_selection_task : public task, public std::enable_shared_fr
         beerocks::eChannelScanErrCode::CHANNEL_SCAN_INVALID_PARAMS;
     eEvent m_dcs_waiting_for_event = eEvent::INVALID_EVENT;
 
-    bool m_is_single_scan_pending = false;
-    bool m_is_single_scan         = false;
+    // The single scan request scan-parameters and scan-results are stored separately than the
```

Comment:
```suggestion
    // The single scan request scan-parameters and scan-results are stored separately from the
```

Reply:
Are there really cases where both scans are triggered simultaneously?
Why don't we handle it by creating two instances of the DCS task?

Reply:
an instance is per interface and not per run
the task is created as part of agent-join and "lives" until agent gets disconnected